src/stores/connectors/eth1/sign.go

```diff
@@ -0,0 +1,294 @@
+	}
+
+	logger.Debug("private transaction signed successfully")
+	return signedRaw, nil
+}
+
+func (c Connector) sign(ctx context.Context, addr string, data []byte) ([]byte, error) {
+	acc, err := c.db.Get(ctx, addr)
+	if err != nil {
+		return nil, err
+	}
+
+	signature, err := c.store.Sign(ctx, acc.KeyID, data, eth1Algo)
+	if err != nil {
+		return nil, err
+	}
+
+	// Recover the recID, please read: http://coders-errand.com/ecrecover-signature-verification-ethereum/
+	for _, recID := range []byte{0, 1} {
+		appendedSignature := append(malleabilityECDSASignature(signature), recID)
+		recoveredPubKey, err := crypto.SigToPub(data, appendedSignature)
+		if err != nil {
+			errMessage := "failed to recover public key candidate with appended recID"
+			c.logger.WithError(err).Error(errMessage, "recID", recID)
+			return nil, errors.InvalidParameterError(errMessage)
+		}
+
+		if bytes.Equal(crypto.FromECDSAPub(recoveredPubKey), acc.PublicKey) {
+			return appendedSignature, nil
+		}
+	}
+
+	return signature, nil
+}
+
+func eeaHash(object interface{}) (hash common.Hash, err error) {
+	hashAlgo := sha3.NewLegacyKeccak256()
+	err = rlp.Encode(hashAlgo, object)
+	if err != nil {
+		return common.Hash{}, err
+	}
+
+	hashAlgo.Sum(hash[:0])
+
+	return hash, nil
+}
+
+// TODO: Delete usage of unnecessary pointers: https://app.zenhub.com/workspaces/orchestrate-5ea70772b186e10067f57842/issues/consensys/quorum-key-manager/96
+func getEncodedPrivateRecipient(privacyGroupID *string, privateFor *[]string) (interface{}, error) {
+	var privateRecipientEncoded interface{}
+	var err error
+	if privacyGroupID != nil {
+		privateRecipientEncoded, err = base64.StdEncoding.DecodeString(*privacyGroupID)
+		if err != nil {
+			return nil, err
+		}
+	} else {
+		var privateForByteSlice [][]byte
+		for _, v := range *privateFor {
+			b, der := base64.StdEncoding.DecodeString(v)
+			if der != nil {
+				return nil, err
+			}
+			privateForByteSlice = append(privateForByteSlice, b)
+		}
+		privateRecipientEncoded = privateForByteSlice
+	}
+
+	return privateRecipientEncoded, nil
+}
+
+func getEIP712EncodedData(typedData *core.TypedData) (string, error) {
+	typedDataHash, err := typedData.HashStruct(typedData.PrimaryType, typedData.Message)
+	if err != nil {
+		return "", err
+	}
+
+	domainSeparatorHash, err := typedData.HashStruct(formatters.EIP712DomainLabel, typedData.Domain.Map())
+	if err != nil {
+		return "", err
+	}
+
+	return fmt.Sprintf("\x19\x01%s%s", domainSeparatorHash, typedDataHash), nil
+}
+
+// Azure generates ECDSA signature that does not prevent malleability
+// A malleable signature can be transformed into a new and valid one for a different message or key.
+// https://docs.microsoft.com/en-us/azure/key-vault/keys/about-keys-details
+// More info about the issue: http://coders-errand.com/malleability-ecdsa-signatures/
+// More info about the fix: https://en.bitcoin.it/wiki/BIP_0062
+func malleabilityECDSASignature(signature []byte) []byte {
```

Comment:
If I understand right, this function makes sure the signature is not malleable?
Can we rename to something more sensible? (makeSignatureNonMalleable() ?)

Reply:
I didn't modify this in this PR so lets not modify it here